src/lib/web.sh

```diff
@@ -76,3 +76,50 @@ function is_html_file()
   [[ "$?" == 0 ]] && return 0
   return 1
 }
+
+# This function recieves a string and converts it to contain only characters that
+# are legal to be used within a URL (https://en.wikipedia.org/wiki/Percent-encoding).
+# Only illegal chars are percent-encoded. A percent-encoding of an ASCII char is
+# its ASCII number in hexadecimal format prefixed by a percent '%'. If the char is
+# non-ASCII, each byte of its byte sequence in UTF-8 is treated as an ASCII char
+# and converted accordingly.
+
+# Note: A full URL shouldn't be passed as argument, as the function will probably
+# break it (e.g., it will convert all foward slashes in the string).
+#
+# @string: String to be encoded
+#
+# Return:
+# The function outputs the encoded string and returns 0 in any case. An empty `string`
+# value is a valid argument.
+#
+# CREDITS:
+# This function was adapted to follow kw's coding style, but it was implemented
+# by meleu (https://github.com/meleu) and it was taken from his blog
+# (https://meleu.sh/urlencode/).
```

Comment:
Where is the source code? Is this code explicitly under GPL or a compatible license?

Reply:
Found that the source reference is https://github.com/dylanaraps/pure-bash-bible#percent-encode-a-string.